File: modules/cubex-core/src/main/java/org/cubexmc/core/Terminable.java
package org.cubexmc.core;

import java.util.Objects;

@FunctionalInterface
public interface Terminable extends AutoCloseable {

    @Override
    void close() throws Exception;

    static Terminable of(Runnable closeAction) {
        Objects.requireNonNull(closeAction, "closeAction");
        return closeAction::run;
    }
}

File: modules/cubex-core/src/main/kotlin/org/cubexmc/core/Terminable.kt
package org.cubexmc.core

import java.util.Objects

fun interface Terminable : AutoCloseable {
    @Throws(Exception::class)
    override fun close()

    companion object {
        @JvmStatic
        fun of(closeAction: Runnable): Terminable {
            Objects.requireNonNull(closeAction, "closeAction")
            return Terminable { closeAction.run() }
        }
    }
}
